dialer_test.go

```diff
@@ -555,6 +550,34 @@ func TestDialerRemovesInvalidInstancesFromCache(t *testing.T) {
 	}
 }
 
+type spyConnectionInfoCache struct {
+	connectInfoError error
+
```

Comment:
nit: is this empty line a typo?

Reply:
just a little way to make the mutex stand out. Let me clean this up in the next PR.